R/html_dependencies.R

```diff
@@ -224,19 +234,36 @@ validate_html_dependency <- function(list) {
   if (is.null(list$version))
     stop("version for html_dependency not provided", call. = FALSE)
   list <- fix_html_dependency(list)
-  if (is.null(list$src$file))
-    stop("path for html_dependency not provided", call. = FALSE)
-  file <- list$src$file
-  if (!is.null(list$package))
-    file <- system.file(file, package = list$package)
-  if (!file.exists(file)) {
-    utils::str(list)
-    stop("path for html_dependency not found: ", file, call. = FALSE)
+  #if (is.null(list$src$file))
+  #  stop("path for html_dependency not provided", call. = FALSE)
+  if (is.null(list$src)) {
+    stop("neither path nor href are provided for html_dependency", call. = FALSE)
+  }
+  if (!is.null(list$src$file)) {
+    file <- list$src$file
+    if (!is.null(list$package))
+      file <- system.file(file, package = list$package)
+    if (!file.exists(file)) {
+      utils::str(list)
+      stop("path for html_dependency not found: ", file, call. = FALSE)
+    }
+  }
+
+  if (!is.null(list$src$href)) {
+    href <- list$src$href
+    if (!RCurl::url.exists(href))
```

Comment:
FWIW we don't do this check in other code paths (like Shiny); if an href is provided, we just assume it'll work. My only reason for pointing this out is because it's possible that the process that's rendering the Rmd might be running on a different machine, with different network connectivity, than the browser that will ultimately be loading the output. (For example, it's pretty common for RStudio Connect or RStudio Server Pro to be run on an airgapped host.)